type-check.ts

```diff
@@ -179,6 +570,19 @@ export function tcStmt(env : GlobalTypeEnv, locals : LocalTypeEnv, stmt : Stmt<n
       } else {
         throw new TypeCheckError("Unbound id: " + stmt.name);
       }
+
+      // TODO: this is an ugly temporary hack for generic constructor
+      // calls until explicit annotations are supported.
+      // Until then constructors for generic classes are properly checked only
+      // when directly assigned to variables and will fail in unexpected ways otherwise.
```

Comment:
Ugh. Understood. Can we throw a big error here if params.length > 0 and we _don't_ know a type to assign it to?

Reply:
I may be misunderstanding the exact condition that you are suggesting that I check here and throw an error. I think I understand what the error/problem is - currently if a Generic Class constructor is used in some unsupported place the params.length part will never get populated, leaving some invalid type tags. Currently our best bet is to hope that it gets caught down the line in some isAssignable check (which may also not always be the case). But I feel like this is not easy to check here in single condition ? -  it would require adding checks in every place that uses and relies on the result of a tcExpr and the type is "constructor" - which I'm not sure if it is worth handling, considering that we plan to get rid of this limitation in the next milestone.

Reply:
Have still not done this since it seems like a lot of work for something we plan to fix as part of the next milestone within the next 2-3 days